src/lib.rs

```diff
@@ -448,7 +448,14 @@ pub mod unsync {
         /// assert_eq!(&*lazy, &92);
         /// ```
         pub fn force(this: &Lazy<T, F>) -> &T {
-            this.cell.get_or_init(|| (this.init)())
+            // Safe because closure is guaranteed to be called at most once
+            // so we only call `F` once, this also guarantees no race conditions
```

Comment:
Interesting! Let's add this to docs of `get_or_init`? Something like this:

> Several threads can call `get_or_init` concurrently, but at most one closure will be executed.

There *can* be an impl that eagerly calls the closure and that uses `.set`. It's useful to say that the current impl guranted not to do that

Reply:
That already seems to be part of the docs!
> ```rust
> ...
> /// Gets the contents of the cell, initializing it with `f`
> /// if the cell was empty. May threads may call `get_or_init`
> /// concurrently with different initializing functions, but
> /// it is guaranteed that only one function will be executed.
> ...
> ```

Reply:
:rofl: I was reading the unsync docs